featuregates/client/pkg/featuregateclient/client.go

```diff
@@ -8,33 +8,34 @@ import (
 	"fmt"
 
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	configv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1"
+	core "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha2"
 	"github.com/vmware-tanzu/tanzu-framework/cli/runtime/config"
 )
 
 // FeatureGateClient defines methods to interact with FeatureGate resources
 type FeatureGateClient struct {
-	c client.Client
+	crClient client.Client
 }
 
-// NewFeatureGateClient returns an instance of FeatureGateClient.
-func NewFeatureGateClient(options ...Option) (*FeatureGateClient, error) {
+// New returns an instance of FeatureGateClient.
+func New(options ...Option) (*FeatureGateClient, error) {
```

Comment:
we have to be careful when changing the names of public functions. We first need to deprecate and remove them later

Reply:
Aren't we going with a whole new API? But I see your point. I'll revert to the longer name.